### ISSUE TITLE
[constructed-inventory] propagating inventory source update to source inventory for constructed inventory

### ISSUE THREAD

**Issue by TheRealHaoLiu** [ansible/awx#13504]

**Comment by AlanCoding** [comment]
A constructed inventories has a relationship to the original inventories that it is built from (called `input_inventories`)

This proposes that, _if the constructed inventory updates_ due to being within its own cache timeout, that any original inventories that have an inventory source _within its own cache timeout_ will also trigger updates.

So let's sketch this out. Say I have original inventory A with an "ec2" inventory source that is set to update-on-launch. A job runs against inventory B, which is a constructed inventory built from inventory A.

So job runs, an update of the constructed inventory source of B is triggered. The crux of this issue is that we may also trigger A's "ec2" source as well.

So (job) -> (inventory A) -> (A's constructed inventory source updates) -> (inventory B) -> (inventory source "ec2" updates)

**Comment by Thulium-Drake** [comment]
@AlanCoding I'm currently running into this issue, is there a workaround that can be used in the mean time?

**Comment by Klaas-** [comment]
> I'm currently running into this issue, is there a workaround that can be used in the mean time?

@Thulium-Drake I just build a workflow that syncs the needed projects/inventories in the correct order and execute it every 15 minutes

**Comment by Thulium-Drake** [comment]
Nice, that'll do for normal runs, but I also use it for a provisioning workflow via Satellite... And you can't have workflows with provisoning webhooks :(

Any ideas?

**Comment by Klaas-** [comment]
> Nice, that'll do for normal runs, but I also use it for a provisioning workflow via Satellite... And you can't have workflows with provisoning webhooks :(
> 
> Any ideas?

I am not familiar with the particular needs of a provisioning webhook but in general if you can start a job in awx you can do whatever you want in Ansible like start a workflow in awx https://docs.ansible.com/ansible/latest/collections/awx/awx/workflow_launch_module.html#ansible-collections-awx-awx-workflow-launch-module for example :)

**Comment by Thulium-Drake** [comment]
Right, sounds like a worthwhile solution to investigate further :)

The main thing for the Foreman/Satellite integration is that it automatically installs a systemd service that will call AWX at first boot. The only thing that needs to happen is that the inventory (constructed in my case) needs to be refreshed so the target freshly provisioned system is actually present.

And as the installation process for an OS takes about 5-10 minutes I now have your earlier suggested workaround run every 5 minutes :zany_face:

**Comment by jon4hz** [comment]
I just stumbeled across this issue and I quite frankly dont understand why a constructed inventory shouldn't also sync its respective input inventories. Usually when you use an external source as inventory you want to most up-to-date state and you only get that by syncing the sources. 

I get that you could use a workflow to achieve this but I think it's a bit unnecessary to create workflows for all the templates when a check box "sync input inventories" in the constructed inventory would do the trick too.

**Comment by Klaas-** [comment]
> I just stumbeled across this issue and I quite frankly dont understand why a constructed inventory shouldn't also sync its respective input inventories. Usually when you use an external source as inventory you want to most up-to-date state and you only get that by syncing the sources.

I don't think anyone is arguing against that, just no one has implemented it. Feel free to create a PR :)
